pkg/executor/infoschema_reader.go

```diff
@@ -927,7 +927,12 @@ func (e *hugeMemTableRetriever) retrieve(ctx context.Context, sctx sessionctx.Co
 
 	if !e.initialized {
 		e.is = sessiontxn.GetTxnManager(sctx).GetTxnInfoSchema()
-		e.dbs = e.extractor.ListSchemas(e.is)
+		dbs, tbls, err := e.extractor.ListSchemasAndTables(ctx, e.is)
```

Comment:
I think the original design is to reduce the memory usage as much as possible. I'm not sure whether this will be a potential problem.

Reply:
Maybe we can add a streaming interface for `SchemaTableInfos` to solve this completely.. Let me revert this change for now.

Reply:
https://github.com/pingcap/tidb/pull/55844/commits/b78dc971e92530ef28efcef3e2de44be59352fa4